### PR TITLE
Integrate deno 1 43 0

### PR DESCRIPTION
**Pull request by andreespirela**

### CODE REVIEW

File: crates/node/ops/os/mod.rs [hunk 1]
```diff
@@ -72,19 +72,6 @@ where
...
-{
-    {
-        let permissions = state.borrow_mut::<P>();
-        permissions.check_sys("abort", "node:process.abort()")?;
-    }
-
-    std::process::abort();
-}
-
 #[op2]
```
Comment (nyannyacha) [inline crates/node/ops/os/mod.rs:75]: It seems to have been reverted.

File: crates/node/ops/process.rs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    state
+        .borrow_mut::<PermissionsContainer>()
+        .check_run_all("process.kill")?;
+    Ok(kill(pid, sig))
+}
+
+#[op2(fast)]
+pub fn op_process_abort() {
+    std::process::abort();
+}
```
Comment (nyannyacha) [inline crates/node/ops/process.rs:64]: It seems to have been reverted.

File: crates/node/polyfills/http.ts [hunk 3]
```diff
@@ -1640,28 +1677,14 @@ export class ServerImpl extends EventEmitter {
...
         });
-
-        tag.fenceRid = fenceRid;
+        // Update socket held by `req`.
+        req.socket = socket;
         this.emit("upgrade", req, socket, Buffer.from([]));
-
-        return internals.RAW_UPGRADE_RESPONSE_SENTINEL;
+        return response;
       } else {
```
Comment (nyannyacha) [inline crates/node/polyfills/http.ts:1688]: This seems to have been reverted as well.

File: crates/sb_os/lib.rs [hunk 4]
```diff
@@ -18,7 +18,7 @@ pub struct MemInfo {
 #[op2]
 #[serde]
 fn op_system_memory_info() -> Option<MemInfo> {
-    let mut mem_info = MemInfo {
+    let mem_info = MemInfo {
         total: 0,
```
Comment (nyannyacha) [inline crates/sb_os/lib.rs:22]:
```suggestion
    let mut mem_info = MemInfo {
        total: 0,
```

File: crates/node/ops/fs.rs [hunk 5]
```diff
@@ -78,3 +79,146 @@ where
...
+    pub blocks: u64,
+    pub bfree: u64,
+    pub bavail: u64,
+    pub files: u64,
+    pub ffree: u64,
+}
+
+#[op2]
+#[serde]
+pub fn op_node_statfs<P>(
```
Comment (laktek) [inline crates/node/ops/fs.rs:97]: I guess this should be a no-op? We don't expose Deno's FS API too.

File: crates/node/ops/mod.rs [hunk 6]
```diff
@@ -7,8 +7,12 @@ pub mod http2;
 pub mod idna;
 pub mod ipc;
 pub mod os;
+pub mod process;
 pub mod require;
 pub mod util;
 pub mod v8;
+pub mod vm;
```
Comment (laktek) [inline crates/node/ops/mod.rs:14]: do we need to expose `vm`? Probably need to stay disabled.

File: crates/node/ops/mod.rs [hunk 7]
```diff
@@ -7,8 +7,12 @@ pub mod http2;
...
+mod vm_internal;
 pub mod winerror;
+pub mod worker_threads;
```
Comment (laktek) [inline crates/node/ops/mod.rs:17]: This should probably staty disabled as well.

File: crates/node/ops/os/mod.rs (hunk 1 above)
Comment (laktek) [reply crates/node/ops/os/mod.rs:75]: @nyannyacha btw, we shouldn't expose `process.abort()` right? or API can be there but it should be a no-op.

File: crates/node/ops/process.rs [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    }
+}
+
+#[op2(fast)]
+pub fn op_node_process_kill(
+    state: &mut OpState,
+    #[smi] pid: i32,
+    #[smi] sig: i32,
+) -> Result<i32, AnyError> {
+    state
```
Comment (laktek) [inline crates/node/ops/process.rs:55]: This should be a no-op.

File: crates/node/ops/vm.rs [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+        let result = self
+            .inner
+            .eval_machine(&mut scope, context)
+            .unwrap_or_else(|| v8::undefined(&mut scope).into());
+        Ok(scope.escape(result))
+    }
+}
+
+#[op2]
+pub fn op_vm_create_script<'a>(
```
Comment (laktek) [inline crates/node/ops/vm.rs:60]: All of vm ops should be no-ops.

File: crates/node/ops/worker_threads.rs [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+    P: NodePermissions + 'static,
+{
+    let resolver = state.borrow::<NpmResolverRc>();
+    let permissions = state.borrow::<P>();
+    resolver.ensure_read_permission(permissions, file_path)
+}
+
+#[op2]
+#[string]
+pub fn op_worker_threads_filename<P>(
```
Comment (laktek) [inline crates/node/ops/worker_threads.rs:29]: Looks like worker threads are disabled. Do we need to add this op?

Comment (andreespirela) [reply crates/node/ops/worker_threads.rs:29]: We could probably remove this file

File: crates/node/ops/vm.rs (hunk 9 above)
Comment (andreespirela) [reply crates/node/ops/vm.rs:60]: Sure

File: crates/node/ops/process.rs (hunk 8 above)
Comment (andreespirela) [reply crates/node/ops/process.rs:55]: Sure

File: crates/node/ops/mod.rs (hunk 7 above)
Comment (andreespirela) [reply crates/node/ops/mod.rs:17]: Perfect

File: crates/node/ops/mod.rs (hunk 6 above)
Comment (andreespirela) [reply crates/node/ops/mod.rs:14]: Sure

File: crates/node/ops/fs.rs (hunk 5 above)
Comment (andreespirela) [reply crates/node/ops/fs.rs:97]: Sure

File: crates/node/ops/os/mod.rs (hunk 1 above)
Comment (nyannyacha) [reply crates/node/ops/os/mod.rs:75]: @laktek 

Exposing this function to a user worker without permission checks leaves open the possibility that malicious code could abort the process.

Making this function to no-op as you say is one way to go.

File: crates/node/polyfills/worker_threads.ts [hunk 11]
```diff
@@ -1,20 +1,53 @@
...
-import { isAbsolute, resolve } from "node:path";
+import { core, internals, primordials } from "ext:core/mod.js";
+import {
+  op_create_worker,
+  op_host_post_message,
+  op_host_recv_ctrl,
+  op_host_recv_message,
+  op_host_terminate_worker,
+  op_message_port_recv_message_sync,
+  op_worker_threads_filename,
```
Comment (nyannyacha) [inline crates/node/polyfills/worker_threads.ts:12]: You removed `op_worker_threads_filename` in rust land, but in js land, call site still exists. Wouldn't this be a problem?
I think there are a few more cases of this.

**Comment by nyannyacha** [comment]
[Blacklisted API](https://github.com/supabase/edge-runtime/blob/0b666d037a3ddffeaac6dca47ffa169a17df287c/crates/base/test_cases/user-worker-san-check/.blacklisted)

**Comment by Garett-MacGowan** [comment]
ETA on this?

File: crates/node/polyfills/os.ts [hunk 12]
```diff
@@ -338,37 +339,37 @@ export function uptime(): number {
...
+  //   _homedir = _homedir ? Buffer.from(_homedir) : _homedir;
+  //   shell = shell ? Buffer.from(shell) : shell;
+  //   username = Buffer.from(username);
+  // }
 
   return {
-    uid,
-    gid,
+    uid: 1000,
+    gid: 1000,
```
[the remark this answers is not in the document]
Comment (andreespirela) [reply crates/node/polyfills/os.ts:369]: Good one

File: crates/node/polyfills/os.ts [hunk 13]
```diff
@@ -188,7 +197,8 @@ export function homedir(): string | null {
       return Deno.env.get("HOME") || null;
     default:
       throw Error("unreachable");
-  }
+  }*/
+  return "/src"
```
[the remark this answers is not in the document]
Comment (andreespirela) [reply crates/node/polyfills/os.ts:201]: Did /src bc of Deno deploy (it's what Deno deploy returns)

[the remark this answers is not in the document]
Comment (laktek) [reply crates/node/polyfills/os.ts:201]: hmm..let's go with `/home/deno` (that's what eszip bundle use as the entrypoint path)

[the remark this answers is not in the document]
Comment (nyannyacha) [reply crates/node/polyfills/os.ts:201]: If that's the return value of deno deploy, I think we're good to go.

**Review by nyannyacha — APPROVED** [review]
LGTM (Thanks for bringing this!)

**Pull request merged**